packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -2067,8 +2082,8 @@ export class MergeTree {
 
 			const wasMovedOnInsert =
 				clientId !== segment.clientId &&
-				segment.seq !== undefined &&
 				seq !== UnassignedSequenceNumber &&
+				segment.seq !== undefined &&
```

Comment:
i'm not convinced this bit is correct (doesn't seem like we should ever be marking segments moved on insert after they are inserted) but in any case this ordering seems to make more sense to me; clientId and seq are checks on the current operation and the latter two are now checks on the segment it affects.